### ISSUE TITLE
ops can not be used as a config entry in place of solids in the Playground run config

### ISSUE THREAD

**Issue by NicolasPA** [dagster-io/dagster#4422]
## Summary
Release [0.12.4](https://github.com/dagster-io/dagster/releases/tag/0.12.4) added: 
> ops can now be used as a config entry in place of solids.

It works in config you provide in the code, but not in the Playground run config.

## Reproduction

Simple example `test_ops.py`:

```python 
from dagster import op, graph, repository


@op
def op1(context, input1: int):
    context.log.info(f"Ran for {input1}")


@graph
def graph1():
    op1()


@repository()
def repo1():
    job1 = graph1.to_job(config={"ops": {"op1": {"inputs": {"input1": 12}}}})
    return [job1]
```

Run: `dagit -f test_ops.py`

![image](https://user-images.githubusercontent.com/22459427/127683805-10145d30-99dc-449d-bb37-0ec63026f6fa.png)

The "Launch Execution" button remains gray and this error is displayed: 
> Received unexpected config entry "ops" at the root. Expected: "{ solids?: { op1: { config?: Any inputs: { input1: (Int | { json: { path: String } pickle: { path: String } value: Int }) } outputs?: [{ result?: { json: { path: String } pickle: { path: String } } }] } } }".

Workarounds:
- leave the Playground run config blank so it uses the one from the code
- replace `ops` with `solids` in the Playground run config.

---
#### Message from the maintainers:
Impacted by this bug? Give it a 👍. We factor engagement into prioritization.

**Comment by sryza** [comment]
@dpeng817

**Comment by dpeng817** [comment]
Thanks for submitting @NicolasPA!

**tl;dr**: Upon further investigation, `ops` does work as a config entry in the playground. The issue is that the default config from the job, which is already satisfied, is re-populating in the playground, and causing a non-helpful error. I have put up an issue to track this [here](https://github.com/dagster-io/dagster/issues/4424).

**More detailed explanation**:
In the code example you posted:
```python
from dagster import op, graph, repository


@op
def op1(context, input1: int):
    context.log.info(f"Ran for {input1}")


@graph
def graph1():
    op1()


@repository()
def repo1():
    job1 = graph1.to_job(config={"ops": {"op1": {"inputs": {"input1": 12}}}})
    return [job1]
```
The config for the job is already satisfied, so the playground expects no additional config. If you try using the playground with empty config, you won't get an error:
<img width="1440" alt="Screen Shot 2021-07-30 at 9 11 45 AM" src="https://user-images.githubusercontent.com/17576880/127703058-648e3f5f-3830-448a-9b06-4622ac0a97da.png">

Likewise, if you remove the default config, and provide the config from the playground, you also wont get an error:
```python
from dagster import op, graph, repository


@op
def op1(context, input1: int):
    context.log.info(f"Ran for {input1}")


@graph
def graph1():
    op1()


@repository()
def repo1():
    job1 = graph1.to_job()
    return [job1]
```
<img width="1440" alt="Screen Shot 2021-07-30 at 9 12 24 AM" src="https://user-images.githubusercontent.com/17576880/127703171-0aff8269-76a1-497c-8975-58d7a10a9b12.png">

However, I noticed that the playground does auto-populate with the default config, which is incorrect. I put up an issue to track this.

**Comment by NicolasPA** [comment]
As a user, it makes sense to me to have the default configuration written in my code to appear in the Playground, so I only have to adjust whatever I want to.

But, I think the real issue that is not addressed by your ticket, is that you can't use "ops" in the Playground if the run config was provided in the repository. I would expect to be able to use "ops" or "solids" indifferently as long as both are allowed in the repository.  
The error clearly indicates that "solids" is expected.

I think this means that when you allowed "ops" to be used in place of "solids" in the run config for release 0.12.4, you forgot to change something in the validation used for the Playground run config, especially in the case when a run config is already provided by the repository.

**Comment by dpeng817** [comment]
Hey @NicolasPA, apologies for the thrash. You are correct that default config should be showing up in the playground for the job.

Turned out to be a bug in the default config buildup; after resolving, the default config no longer errors from the playground, as expected:
<img width="1440" alt="Screen Shot 2021-07-31 at 8 48 58 AM" src="https://user-images.githubusercontent.com/17576880/127749797-ee0c0bfa-b8d3-4132-8cbb-413cf9cc24b1.png">


Thanks again for surfacing! A fix should be out for next release.

One thing to note, an unfortunate side affect of allowing for both config entries is that (for now), config errors related to op config are surfaced in terms of solid.

**Comment by NicolasPA** [comment]
Thank you for the fix @dpeng817, looking forward to it!

**Comment by dpeng817** [comment]
Closed by https://github.com/dagster-io/dagster/commit/3751d986cf7025997044b4e813ce36be7df50dd4
